### ISSUE TITLE
Compile generated UnsafeAccessors classes don't match underlying class accessibility level (Microsoft.EntityFrameworkCore.Tasks 9.0.0-rc.2.24474.1)

### ISSUE THREAD

**Issue by NicholasMoss** [dotnet/efcore#35013]
When using `EFOptimizeContext` for build-time optimisation and your have an Entity which is less accessible than `public` the generated `UnsafeAccessors` class will raise inconsistent accessibility errors.

For example even if the source class is internal the generated UnsafeAccessors class will look something like this:

```C#
namespace Financials.Data.Read.Common
{
    public static class TestEntityUnsafeAccessors
    {
        [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "<TestProperty>k__BackingField")]
        public static extern ref string TestProperty(TestEntity @this);
    }
}
```

Where instead the accessibility of TestEntityUnsafeAccessors should be internal to match the source.

**Comment by qhy040404** [comment]
`public static class ` is hard-coded here and is completely unreasonable. And most places (maybe even all places) in the generator of the entire project are directly hard-coded `public`.

https://github.com/dotnet/efcore/blob/08f094b6823df39d5b2894dbc64f90143e917259/src/EFCore.Design/Scaffolding/Internal/CSharpRuntimeModelCodeGenerator.cs#L216
